### PR TITLE
Add MQTT Number

### PR DESCRIPTION
**Pull request by dgomes**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/44739
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/number.mqtt.markdown [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+title: "MQTT Number"
+description: "Instructions on how to interact with a device exposing a Number through MQQT from within Home Assistant."
```
Comment (JeffLIrion) [inline source/_integrations/number.mqtt.markdown]: Typo: MQQT

File: source/_integrations/number.mqtt.markdown [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+number:
+  - platform: mqtt
+    command_topic: my-device/threshold
+```
+
+{% configuration %}
+availability:
+  description: A list of MQTT topics subscribed to receive availability (online/offline) updates. Must not be used together with `availability_topic`.
+  required: false
+  type: list
```
Comment (JeffLIrion) [inline source/_integrations/number.mqtt.markdown:28]: Given that you list keys, shouldn't this be a map rather than a list?

Comment (dgomes) [reply source/_integrations/number.mqtt.markdown:28]: I must confess this was copy & paste from existing mqtt documentation 😬

Comment (emontnemery) [reply source/_integrations/number.mqtt.markdown:28]: It's a list of maps, the existing documentation is correct.

**Comment by emontnemery** [comment]
I think you forgot to document the `state_topic`. Please also add the same change as here: https://github.com/home-assistant/home-assistant.io/pull/16158

**Review by emontnemery — APPROVED** [review]
Looks good!

**Pull request merged**